batchflow/tests/pipeline_init_model_test.py

```diff
@@ -0,0 +1,55 @@
+""" Tests for pipeline init_model action """
+# pylint: disable=import-error, no-name-in-module
+import pytest 
+
+from batchflow import DatasetIndex, Pipeline, Batch, C
+from batchflow.models import BaseModel
+from batchflow.model_dir import NonInitializedModel
+
+@pytest.fixture
+def dataset():
+    """ Toy dataset """
+    index = DatasetIndex(100)
+    return Dataset(index, Batch)
+
+@pytest.mark.parametrize('use_algebra', [True, False])
+@pytest.mark.parametrize('config', [dict(mode='static', base_class=BaseModel, name='model', config={}),
+                                    dict(mode='dynamic', base_class=BaseModel, name='model', config={})])
+def test_single_model(use_algebra, config, dataset):
+    """ Verifies that the model in init_model is initialized as expected with config provided, i.e.
+    pipeline contains the model with given name belongs to given class.
+    Also veriying methods of linking config to the pipeline: via pipeline algebra and manually. """
+    if not use_algebra:
+        pipeline = dataset.pipeline(config).init_model(C('mode'), C('base_class'), C('name'), C('config'))
+    else:
+        pipeline = dataset.pipeline().init_model(C('mode'), C('class'), C('base_name'), C('config')) << config
+    pipeline.next_batch(1)
+
+    if config['mode'] == 'static': 
+        assert type(pipeline.m(config['name']) == config['class']
+    else:
+        assert type(pipeline.m(config['name']) == NonInitializedModel
+
+@pytest.mark.parametrize('use_algebra1', [True, False])
+@pytest.mark.parametrize('use_algebra2', [True, False])
+@pytest.mark.parametrize('mode', ['static', 'dynamic'])
+@pytest.mark.parametrize('config1', 'config2' [(dict(class=BaseModel, name='model1', config={}),
+                                                dict(class=BaseModel, name='model2', config={})])
+def test_multiple_models(dataset, use_algebra1, use_algebra2, mode,  config1, config2):
+    """ Verifies that several modeles can be initialized and stored in the pipeline via sequentially linking
+    different configs to the pipeline. """
+    config1.update(mode=mode)
```

Comment:
all what is going on below is extremely confusing

Reply:
> all what is going on below is extremely confusing

The tests provided essentially checks two things, which are currently not working in `master` 

1.  Args passed to `init_model` as C() expressions are being properly unpacked inside the method.
2. Sequentially linking configs with different models to the same pipeline with `init_model` method results in properly storing all the models.

Reply:
If the code needs explanation it has to be rewritten. No explanation is needed besides the code.